src/scripts/main.js

```diff
@@ -1,3 +1,53 @@
 'use strict';
 
-// write code here
+// Функция для сортировки сотрудников по зарплате
+const sortEmployeesBySalary = function (selector) {
+  // Получаем все элементы списка по переданному селектору
+  const employeeItems = document.querySelectorAll(selector);
+
+  // Преобразуем NodeList в массив и сортируем его по зарплате
+  const sortedEmployee = Array.from(employeeItems).sort((a, b) => {
+    // Извлекаем зарплату из атрибутов данных и преобразуем в число
+    const salaryA = parseInt(a.dataset.salary.replace(/[$,]/g, ''), 10);
+    const salaryB = parseInt(b.dataset.salary.replace(/[$,]/g, ''), 10);
+
+    return salaryB - salaryA; // Сортируем по убыванию
+  });
+
+  return sortedEmployee; // Возвращаем отсортированный массив сотрудников
+};
+
+// Функция для получения массива объектов сотрудников
+const getEmployeeArray = function (selector) {
+  // Получаем все элементы списка по переданному селектору
+  const employeeItems = document.querySelectorAll(selector);
+
+  // Преобразуем NodeList в массив объектов
+  const employees = Array.from(employeeItems).map((item) => ({
+    name: item.textContent.trim(),
+    position: item.dataset.position,
+    salary: item.dataset.salary,
+    age: parseInt(item.dataset.age, 10),
+  }));
+
+  return employees; // Возвращаем массив сотрудников
```

Comment:
The function `getEmployeeArray` is currently unused. If you meant to use this function, make sure to replace the relevant parts of your code with a call to this function.